src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -214,6 +220,24 @@ private ValueTask<HttpConnection> GetConnectionAsync(HttpRequestMessage request,
                 new HttpConnectionWithFinalizer(this, stream, transportContext); // finalizer needed to signal the pool when a connection is dropped
         }
 
+        private async ValueTask<Stream> EstablishProxyTunnel(CancellationToken cancellationToken)
+        {
+            // Send a CONNECT request to the proxy server to establish a tunnel.
+            HttpRequestMessage tunnelRequest = new HttpRequestMessage(new HttpMethod("CONNECT"), _proxyUri);
```

Comment:
Rather than allocating a new HttpMethod object each time, let's cache it.  You could just add an `internal` one to https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpMethod.cs#L21 (and then it could potentially be exposed publicly later if desired).